101-event-grid-resource-events-to-webhook/azuredeploy.json

```diff
@@ -14,27 +14,48 @@
             "metadata": {
                 "description": "The URL for the WebHook to receive events. Create your own endpoint for events."
             }
+        },
+        "systemTopicName": {
+            "defaultValue": "systemTopicName",
+            "type": "String",
+            "metadata": {
+                "description": "The name of the topic to create."
+            }
         }
     },
     "resources": [
         {
-            "type": "Microsoft.EventGrid/eventSubscriptions",
-            "name": "[parameters('eventSubName')]",
-            "apiVersion": "2018-01-01",
+            "type": "Microsoft.EventGrid/systemTopics",
+            "apiVersion": "2020-04-01-preview",
+            "name": "[parameters('systemTopicName')]",
+            "location": "global",
+            "properties": {
+                "source": "[concat('/subscriptions/', subscription().subscriptionId)]",
```

Comment:
I think this is the change that's causing the problem... it's trying to elevate to subscription scope for some operation and the principal deploying only has permission to the resourceGroup - what's the intent of this change?

@tfitzmac - is this a doc sample? (where we care about the change) it's not marked as such in metadata but I think this sample pre-dates that approach

Reply:
The goal is to fix the issue mentioned. The previous template does not seem to deploy event grid topic for subscriptions but only for resource groups.

Reply:
ok, the readme says "resource group or a subscription" - so if we can use the same template for either, then there should be a param for determining which should be done... and then in CI we'd choose the RG path.  If it's only for a subscription, then this needs to be moved to a subscription deployment since perms are needed at that scope to deploy it.

Still want @tfitzmac to chime in on how the sample is used in docs (if so)

Reply:
Yes, the doc owner should be spelluru

Reply:
Actually, it is referenced with the "deploy to Azure" button in the docs, but the code block is not displayed. In my opinion, we shouldn't add a doc owner for those cases, because the writer doesn't really need to be looped in for every change. However, I could be convinced otherwise.